endpoints/v2/blob.py

```diff
@@ -389,6 +399,11 @@ def upload_chunk(namespace_name, repo_name, upload_uuid):
 @anon_protect
 @check_readonly
 def monolithic_upload_or_last_chunk(namespace_name, repo_name, upload_uuid):
+    # Check if pushes are disabled on the registry
+    if app.config.get("DISABLE_PUSHES", False):
+        logger.debug("Cannot push images, pushes disabled on registry.")
+        raise PushesDisabled
```

Comment:
You'll need to also check the other non-GET endpoints no?
Couldn't you add this check to the existing `check_readonly` decorator?

Reply:
The check is added to all other non-GET endpoints: it's added to the `POST` and `PATCH`. Direct blob deletions are not possible anyway, and for blob requests that have an upload id it makes no sense to add the check since it's obvious that pushes are enabled there. I didn't however add the check to the manifest `DELETE` request which would delete the manifest and subsequent blobs. I think I need to add it there as well.

I was thinking about the decorator too, I'll revise that part.